Dockerfile.alpine

```diff
@@ -0,0 +1,22 @@
+FROM golang:1.11-alpine3.8 AS builder
+
+WORKDIR $GOPATH/src/github.com/nats-io/gnatsd
+
+MAINTAINER Waldemar Quevedo <wally@synadia.com>
+
+RUN apk add --update git
+
+COPY . .
+
+RUN CGO_ENABLED=0 go build -v -a -tags netgo -installsuffix netgo -ldflags "-s -w -X github.com/nats-io/gnatsd/server.gitCommit=`git rev-parse --short HEAD`" -o /gnatsd
+
+FROM alpine:3.8
+
+RUN apk add --update ca-certificates
+
+COPY --from=builder /gnatsd /gnatsd
```

Comment:
I think we should create a dir, nats I think, have bin and conf dirs. sym link /nats/bin/gnatsd to /bin

Reply:
Sorry for the delay, I have applied the changes and this is how it looks now:

```sh
$ docker run --entrypoint /bin/sh -it wallyqs/nats-server:edge
...
/ # tree /nats/
/nats/
├── bin
│   └── gnatsd
└── conf
    └── gnatsd.conf

2 directories, 2 files
/ # ls -la /bin/ | grep gnatsd -A1 -B1
lrwxrwxrwx    1 root     root            12 Sep 11 20:23 getopt -> /bin/busybox
lrwxrwxrwx    1 root     root            16 Dec 21 17:31 gnatsd -> /nats/bin/gnatsd
lrwxrwxrwx    1 root     root            12 Sep 11 20:23 grep -> /bin/busybox
/ # ls -la /nats/*/*
-rwxr-xr-x    1 root     root       8932640 Dec 21 17:30 /nats/bin/gnatsd
-rw-r--r--    1 root     root           671 Dec 21 17:05 /nats/conf/gnatsd.conf

$ docker run wallyqs/nats-server:edge
[1] 2018/12/21 17:43:36.979863 [INF] Starting nats-server version 2.0.0-beta.12
[1] 2018/12/21 17:43:36.979973 [INF] Git commit [2bc515a]
[1] 2018/12/21 17:43:36.980124 [INF] Starting http monitor on 0.0.0.0:8222
[1] 2018/12/21 17:43:36.980349 [INF] Listening for client connections on 0.0.0.0:4222
[1] 2018/12/21 17:43:36.980404 [INF] Server id is NDEHGDUGIFECN7NUZUPOEDEDVOGAPEWHQSPEGGQNSMIZFVIO2MPZSURN
[1] 2018/12/21 17:43:36.980429 [INF] Server is ready
[1] 2018/12/21 17:43:36.981009 [INF] Listening for route connections on 0.0.0.0:6222
^C[1] 2018/12/21 17:43:37.816006 [INF] Server Exiting..
```

Reply:
ok getting closer. We should symlink the binary to /bin/nats-server IMO.
Also move gnatsd to nats-server in prep for rename.

Reply:
How would a user easily provide their own config to run this docker image? Is it easy?

Reply:
Sounds good on renaming, will make the changes.

About custom config, since we are using alpine then users could do something like this too now without having to mount volumes for example:
```
docker run --entrypoint /bin/sh wallyqs/nats-server:edge-2 -c '
cat > /nats/conf/gnatsd.conf <<CONF
  listen = 4222
  authorization {
    user = foo
    pass = bar
  }   
CONF
/bin/gnatsd -c /nats/conf/gnatsd.conf
'
```

Reply:
What if they had one in a file, is that fairly straightforward as well with volume mount? Does that override ours? Can they override ports too etc?

Reply:
Made the rename changes and also added another symlink to `/gnatsd` so that the image is backwards compatible with the original `FROM scratch` image which had binary at root since some tooling may have relied on that.

```
/ # tree /nats
/nats
├── bin
│   └── nats-server
└── conf
    └── nats.conf

2 directories, 2 files
/ # ls -la /bin/* | grep nats -A1 -B1
lrwxrwxrwx    1 root     root            12 Sep 11 20:23 /bin/getopt -> /bin/busybox
lrwxrwxrwx    1 root     root            21 Dec 21 19:50 /bin/gnatsd -> /nats/bin/nats-server
lrwxrwxrwx    1 root     root            12 Sep 11 20:23 /bin/grep -> /bin/busybox
--
lrwxrwxrwx    1 root     root            12 Sep 11 20:23 /bin/mv -> /bin/busybox
lrwxrwxrwx    1 root     root            21 Dec 21 19:50 /bin/nats-server -> /nats/bin/nats-server
lrwxrwxrwx    1 root     root            12 Sep 11 20:23 /bin/netstat -> /bin/busybox
/ # ls -la /gnatsd 
lrwxrwxrwx    1 root     root            21 Dec 21 19:50 /gnatsd -> /nats/bin/nats-server
/ # nats-server 
[13] 2018/12/21 19:51:13.668498 [INF] Starting nats-server version 2.0.0-beta.12
^C[13] 2018/12/21 19:51:14.082594 [INF] Server Exiting..

/ # /gnatsd
[21] 2018/12/21 19:51:20.598620 [INF] Starting nats-server version 2.0.0-beta.12
```

Reply:
That seems correct to me.

Reply:
Using volumes then yes would also be possible to override with a local file, for example:

```sh
$ echo '
  listen = 4333
' > my-nats.conf

$ docker run -v $(pwd)/my-nats.conf:/nats/conf/nats.conf -it wallyqs/nats-server:edge
[1] 2018/12/21 20:12:19.113881 [INF] Starting nats-server version 2.0.0-beta.12
[1] 2018/12/21 20:12:19.114099 [INF] Listening for client connections on 0.0.0.0:4333
```

Reply:
ok that looks good. Should we call it nats-server.conf?

Reply:
sgtm!

Reply:
ok updated so that it is now `nats-server.conf`